centreon-dsm/www/modules/centreon-dsm/core/configuration/services/formSlot.php

```diff
@@ -38,198 +38,212 @@
-        )
-    );
-    $form->setDefaults($pool);
-    $form->freeze();
-} elseif ($o == "c") {
-    // Modify a pool information
-    $subC = $form->addElement('submit', 'submitC', _("Save"), array("class" => "btc bt_success"));
-    $res = $form->addElement('reset', 'reset', _("Reset"), array("class" => "btc bt_default"));
-    $form->setDefaults($pool);
-} elseif ($o == "a") {
-    // Add a pool information
-    $subA = $form->addElement('submit', 'submitA', _("Save"), array("class" => "btc bt_success"));
-    $res = $form->addElement('reset', 'reset', _("Reset"), array("class" => "btc bt_default"));
-}
+    if ($o == "w") {
+        // Just watch a pool information
+        $form->addElement(
+            "button",
+            "change",
+            _("Modify"),
+            array(
+                "class" => "btc bt_default",
+                "onClick" => "javascript:window.location.href='?p=" . $p . "&o=c&pool_id=" . $pool_id . "'"
+            )
+        );
+        $form->setDefaults($pool);
+        $form->freeze();
+    } elseif ($o == "c") {
+        // Modify a pool information
+        $subC = $form->addElement('submit', 'submitC', _("Save"), array("class" => "btc bt_success"));
+        $res = $form->addElement('reset', 'reset', _("Reset"), array("class" => "btc bt_default"));
+        $form->setDefaults($pool);
+    } elseif ($o == "a") {
+        // Add a pool information
+        $subA = $form->addElement('submit', 'submitA', _("Save"), array("class" => "btc bt_success"));
+        $res = $form->addElement('reset', 'reset', _("Reset"), array("class" => "btc bt_default"));
+    }
 
-$valid = false;
-$msgErr = "";
-if ($form->validate() && $from_list_menu == false) {
-    $poolObj = $form->getElement('pool_id');
-    if ($form->getSubmitValue("submitA")) {
-        try {
-            $pId = insertpoolInDB();
-            $valid = true;
-            $poolObj->setValue($pId);
-        } catch (Exception $e) {
-            $valid = false;
-            $msgErr = $e->getMessage();
-        }
-    } elseif ($form->getSubmitValue("submitC")) {
-        try {
-            $valid = updatePoolInDB($poolObj->getValue());
-        } catch (Exception $e) {
-            $valid = false;
-            $msgErr = $e->getMessage();
+
+    if ($form->validate() && $from_list_menu == false) {
+        $poolObj = $form->getElement('pool_id');
+        if ($form->getSubmitValue("submitA")) {
+            try {
+                $pId = insertpoolInDB();
+                $valid = true;
+                $poolObj->setValue($pId);
+            } catch (Exception $e) {
+                $valid = false;
+                $msgErr = $e->getMessage();
+            }
+        } elseif ($form->getSubmitValue("submitC")) {
+            try {
+                $valid = updatePoolInDB($poolObj->getValue());
+            } catch (Exception $e) {
+                $valid = false;
+                $msgErr = $e->getMessage();
+            }
```

Comment:
You do a global catch so these exceptions must not be catch here to have a user friendly message ;)

Reply:
done

Reply:
thanks